scripts/xapi-tracing-log-trim.cron

```diff
@@ -0,0 +1,3 @@
+# Run at 20 minutes past the hour,
```

Comment:
Could you use logrotate here? http://www.skrenta.com/rt/man/logrotate.8.html

That is already set up to run on a schedule and can rotate/compress based on file size.
Although that may only work well if a single well known filename is used, you can use '*' in logrotate, but I'm not sure whether that would actually delete old files completely or you'd have to do it by hand.

Reply:
I could look into doing that

Reply:
Another option is https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html which can clean up files in a directory based on age.

Reply:
I think tmpfiles.d would. For logrotate you have to use wildcards, but you'll have to do some tests to see how it works.

Reply:
Rotating logfiles has the problem that you might want to rotate a file that is currently open - assuming the logging mechanism keeps the file open between writes. How is this handled? It could be an option to log through the syslog service to avoid this problem rather than writing to a regular file.

Reply:
for now maybe keep this script if neither tmpfiles.d or syslog is appropriate. Syslog would have all kinds of limit on line length and rate limiting, might be more difficult to use.

Reply:
(also if we want to switch to a binary format later syslog wouldn't work). To change to something else signals could be used if needed (or the application can be mindful of the size of the file it has already written and reopen a new file on its own)

Reply:
After the latest changes, I can test the script successfully in dom0 using eg.
```
MAX_MB=10 /opt/xensource/libexec/xapi-tracing-log-trim.sh  #remove the oldest files beyond 10MB total in dir
MAX_DAYS=0 /opt/xensource/libexec/xapi-tracing-log-trim.sh #remove all the files immediately
```